src/pages/task/actionButtons/PreviousCommits.tsx

```diff
@@ -0,0 +1,212 @@
+    GetBaseVersionAndTaskQuery,
+    GetBaseVersionAndTaskQueryVariables
+  >(GET_BASE_VERSION_AND_TASK, {
+    variables: { taskId },
+  });
+
+  const [fetchParentTask] = useLazyQuery<
+    GetLastMainlineCommitQuery,
+    GetLastMainlineCommitQueryVariables
+  >(GET_LAST_MAINLINE_COMMIT, {
+    onCompleted: (data) => {
+      const task = getTaskFromMainlineCommitsQuery(data);
+      if (task) setParentTask(task);
+    },
+  });
+
+  const [fetchLastPassing] = useLazyQuery<
+    GetLastMainlineCommitQuery,
+    GetLastMainlineCommitQueryVariables
+  >(GET_LAST_MAINLINE_COMMIT, {
+    onCompleted: (data) => {
+      const task = getTaskFromMainlineCommitsQuery(data);
+      if (task) setLastPassingTask(task);
+    },
+  });
+
+  const [fetchLastExecuted] = useLazyQuery<
+    GetLastMainlineCommitQuery,
+    GetLastMainlineCommitQueryVariables
+  >(GET_LAST_MAINLINE_COMMIT, {
+    onCompleted: (data) => {
+      const task = getTaskFromMainlineCommitsQuery(data);
+      if (task) setLastExecutedTask(task);
+    },
+  });
+
+  const { baseTask, versionMetadata, buildVariant, displayName } =
+    taskData?.task ?? {};
+  const { projectIdentifier, order } = versionMetadata?.baseVersion ?? {};
+  const skipOrderNumber = versionMetadata?.isPatch ? order + 1 : order;
+  const bvOptionsBase = {
+    tasks: [applyStrictRegex(displayName)],
+    variants: [applyStrictRegex(buildVariant)],
+  };
+
+  // Hook to determine the parent task. If mainline commit, use fetchParentTask function to get task from
+  // previous mainline commit. Otherwise, just extract the base task from the task data.
+  useEffect(() => {
+    if (versionMetadata?.isPatch === false) {
+      fetchParentTask({
+        variables: {
+          projectIdentifier,
+          skipOrderNumber,
+          buildVariantOptions: {
+            ...bvOptionsBase,
+          },
+        },
+      });
+    } else if (versionMetadata?.isPatch === true) {
+      setParentTask(baseTask);
+    }
+  }, [versionMetadata]); // eslint-disable-line react-hooks/exhaustive-deps
+
+  // Hook to find last passing & last executed tasks. This is a separate hook because the parent task must be
+  // known before proceeding.
+  useEffect(() => {
+    if (parentTask) {
+      if (parentTask.status !== TaskStatus.Succeeded) {
+        fetchLastPassing({
+          variables: {
+            projectIdentifier,
+            skipOrderNumber,
+            buildVariantOptions: {
+              ...bvOptionsBase,
+              statuses: [TaskStatus.Succeeded],
+            },
+          },
+        });
+      }
+      if (!isFinishedTaskStatus(parentTask.status)) {
+        fetchLastExecuted({
+          variables: {
+            projectIdentifier,
+            skipOrderNumber,
+            buildVariantOptions: {
+              ...bvOptionsBase,
+              statuses: finishedTaskStatuses,
+            },
+          },
+        });
+      }
+    }
+  }, [parentTask]); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
Could we avoid making these queries if the user never clicks on these buttons? I think the queries are fast enough where we can avoid making them preemptively. 
 Base commit/ parent commit make sense to me since they are the default values/ default state of the button.
But the other ones may never be hit so maybe it would make sense to only call the query if the user selects one of them.